src/components/ComponentRelation/useNudgeData.ts

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import { useComponents } from '../../hooks/useComponents';
+import { ComponentKind, NudgeStats } from '../../types';
+import { useWorkspaceInfo } from '../../utils/workspace-context-utils';
+import { ComponentRelationNudgeType, ComponentRelationValue } from './type';
+
+export const useNudgeData = (application: string): [ComponentRelationValue[], boolean, any] => {
+  const { namespace } = useWorkspaceInfo();
+  const [components, loaded, error] = useComponents(namespace, application);
+  const nudgeData: ComponentRelationValue[] = React.useMemo(() => {
+    return loaded && !error
+      ? components.reduce((acc, val: ComponentKind) => {
+          if (val.spec?.[NudgeStats.NUDGES]) {
+            const data: ComponentRelationValue = {
+              source: val.metadata.name,
+              nudgeType: ComponentRelationNudgeType.NUDGES,
+              target: val.spec[NudgeStats.NUDGES],
```

Comment:
we can take into account that only available components in the application will be used to show relationships.

```suggestion
              target: val.spec[NudgeStats.NUDGES]?.filter(
                (cname) => !!components?.find((co) => co?.metadata?.name === cname),,
```

Reply:
This restriction to limit nudges within an Application have actually been loosened. We should no longer filter to only Components in the one Application. 

@MariaLeonova , this has a potential adverse effect on the UX as the number of components can be much larger. 

Would it be reasonable/possible to be able to filter components based on their Application? Or to have Components listed with their Application name (i.e. `<Application>/<Component>`) for grouping "similar" Components together?

Reply:
@arewm We will add support for component relations across applications in a new story.

Reply:
@sahil143 , do you have the feature to for cross-Application support?